### ISSUE TITLE
KMC installation problem

### ISSUE THREAD

**Issue by wjj666** [refresh-bio/KMC#181]
Hi,
I installed kmc with `conda install -c bioconda kmc` (v 3.2.1). When running kmc, it reported this error:
```
FATAL: kernel too old
Aborted
```
I also tried installed kmc with `conda install -c bioconda/label/cf201901 kmc`, it reported this error:
`kmc: /lib64/libc.so.6: version GLIBC_2.14 not found`

The GCC version:
```
 $ gcc --version
gcc (GCC) 5.2.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

The as version:
 ```
$ as --version
GNU assembler version 2.20.51.0.2-5.44.el6 20100205
Copyright 2009 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `x86_64-redhat-linux'.
```

Could you please help me solve this problem? Thank you very much.

**Comment by marekkokot** [comment]
Hi,

this is new to me, I've never seen "kernel too old" message. 
Out of curiosity, what is your kernel version? You may check it with
```
uname -r
```
Your g++ version seems to be quite old, but maybe kmc can be compiled with it. Have you tried compiling from sources?
We would really like to make KMC work on each Linux distribution, this is why we link it with ```-static``` flag.
The version of KMC installed with
```
conda install -c bioconda/label/cf201901 kmc
```
was not prepared on bioconda by us and I am not sure whats it going on there, but it seems (from what kind of error you get) like it's not using a statically compiled version.
Nevertheless, if you show me your kernel version maybe I will be able to install it on a virtual machine and reproduce this behavior and maybe fix it or find some suggestions for you.
As I understand, upgrading to a newer kernel is not an option for you?

Anyway, thanks for using KMC and reporting this bug.

Best,
Marek

**Comment by wjj666** [comment]
Hi, this is the kernel information:
```
 $ uname -r
2.6.32-642.el6.x86_64
```

**Comment by marekkokot** [comment]
Hi, 

honestly, it seems this is a really old kernel. What version of RHEL are you running? I was tried to use some old Debian version to test such an old kernel, but there are soooooo many issues to use it that I resigned.
Have you tried compiling from sources?
```
git clone https://github.com/refresh-bio/KMC/
cd KMC
make 
```
Out of curiosity why is it not possible for you to upgrade your system?

I was able to compile kmc with g++ 5.4.0 which results in executable for kernel 2.6.32, so it should work for you.
I'm attaching a compiled version
[KMC3.2.1.linux.for-kernel-2.6.32.tar.gz](https://github.com/refresh-bio/KMC/files/7884834/KMC3.2.1.linux.for-kernel-2.6.32.tar.gz)

Let me know if it works. 

I prefer not to use this version as an official release because I am not sure how it will influence running time (I am not expecting any differences, but as speed is one of the most important KMC advantages I don't want the risk of officially publishing the version that is slower just because it is compiled with the older compiler).

For future me: I have used docker image ```sudo docker run -it --rm conanio/gcc5 /bin/bash``` to easily get gcc5 compiler.
